lib/include/private/iot_mqtt_internal.h

```diff
@@ -372,9 +372,7 @@ typedef struct _mqttConnection
     const IotNetworkInterface_t * pNetworkInterface; /**< @brief Network interface provided to @ref mqtt_function_connect. */
     IotMqttCallbackInfo_t disconnectCallback;        /**< @brief A function to invoke when this connection is disconnected. */
 
-    #if IOT_MQTT_ENABLE_SERIALIZER_OVERRIDES == 1
-        const IotMqttSerializer_t * pSerializer; /**< @brief MQTT packet serializer overrides. */
-    #endif
+    const IotMqttSerializer_t * pSerializer; /**< @brief MQTT packet serializer overrides. */
```

Comment:
Adding an additional pointer to this structure when not using the serializer overrides; this allows the code to be simplified quite a bit using the new `IOT_MQTT_SERIALIZER_OVERRIDE_SELECTOR` macro.